pkgs/development/libraries/libdicom/default.nix

```
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, uthash
+, meson
+, ninja
+, cmake
+, pkg-config
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "libdicom";
+  version = "1.0.5";
+
+  src = fetchFromGitHub {
+    owner = "ImagingDataCommons";
+    repo = "libdicom";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-9n0Gp9+fmTM/shgWC8zpwt1pic9BrvDubOt7f+ZDMeE=";
+  };
+
+  buildInputs = [ uthash ];
+
+  nativeBuildInputs = [ meson ninja cmake pkg-config ];
+
+  mesonBuildType = "release";
+
+  mesonFlags = [ "-Dtests=false" ];
```

```suggestion
  mesonFlags = lib.optionals (!finalAttrs.doCheck) [ "-Dtests=false" ];
```

If achievable we'd like to run tests. Try adding `check` back to `nativeCheckInputs` and set `doCheck = true;`

---

https://github.com/NixOS/nixpkgs/pull/271241#issuecomment-1834596148

---

saw that only afterwards

---

It happens with you use `pkgCross`. I noticed you don't set `doCheck`, but when my suggestions are applied then this works:

```
nix-repl> libdicom.mesonFlags                                 
[ ]

nix-repl> pkgsCross.aarch64-multiplatform.libdicom.mesonFlags
[ "--cross-file=/nix/store/xvvrwznyp55hlq33m4hfighkbzvbqw9s-cross-file.conf" "-Dtests=false" ]
```

---

i.e. `nix-build . -A pkgsCross.aarch64-multiplatform.libdicom`